numpy/lib/function_base.py

```diff
@@ -1854,15 +1855,30 @@ def cov(m, y=None, rowvar=1, bias=0, ddof=None):
         is transposed: each column represents a variable, while the rows
         contain observations.
     bias : int, optional
-        Default normalization is by ``(N - 1)``, where ``N`` is the number of
-        observations given (unbiased estimate). If `bias` is 1, then
-        normalization is by ``N``. These values can be overridden by using
-        the keyword ``ddof`` in numpy versions >= 1.5.
+        Default normalization is by ``(N - 1)``, where ``N`` corresponds to the
+        number of observations given (unbiased estimate). If `bias` is
+        1, then normalization is by ``N``. These values can be overridden by
+        using the keyword ``ddof`` in numpy versions >= 1.5.
     ddof : int, optional
         .. versionadded:: 1.5
-        If not ``None`` normalization is by ``(N - ddof)``, where ``N`` is
-        the number of observations; this overrides the value implied by
-        ``bias``. The default value is ``None``.
+        If not ``None`` normalization is by ``(N - ddof)``, where ``N``
+        corresponds to the number of observations given; this overrides the
+        value implied by ``bias``. The default value is ``None``.
+    frequencies : array_like, int, optional
+        An array of integer frequencies associated with the values in `m` and
+        `y`. The frequencies array is 1-D and its length must be equal to the
+        size of `m` and `y` along the specified axis. If `frequencies=None`
+        all data in `m` and `y` are assumed to have a frequency equal to one.
+    weights : array_like, optional
+        An array of weights associated with the values in `m` and `y`
```

Comment:
Lets get rid of all the latex up here, it should be in the Notes section instead. Suggested signature and docstring maybe something like
```
def cov(m, y=None, rowvar=1, bias=0, ddof=None, fweight=None, aweight=None):
    """
    Estimate a covariance matrix, given data and weights.

    Covariance indicates the level to which two variables vary together.
    If we examine N-dimensional samples, :math:`X = [x_1, x_2, ... x_N]^T`,
    then the covariance matrix element :math:`C_{ij}` is the covariance of
    :math:`x_i` and :math:`x_j`. The element :math:`C_{ii}` is the variance
    of :math:`x_i`.

    See the notes for a mathematically explicit explanation.

    Parameters
    ----------
    m : array_like
        A 1-D or 2-D array containing multiple variables and observations.
        Each row of `m` represents a variable, and each column a single
        observation of all those variables. Also see `rowvar` below.
    y : array_like, optional
        An additional set of variables and observations. `y` has the same
        form as that of `m`.
    rowvar : int, optional
        If `rowvar` is non-zero (default), then each row represents a
        variable, with observations in the columns. Otherwise, the relationship
        is transposed: each column represents a variable, while the rows
        contain observations.
    bias : int, optional
        Default normalization is by ``(N - 1)``, where ``N`` is the number of
        observations given (unbiased estimate). If `bias` is 1, then
        normalization is by ``N``. These values can be overridden by using
        the keyword ``ddof`` in numpy versions >= 1.5.
    ddof : int, optional
        .. versionadded:: 1.5
        If not ``None`` the default value implied by `bias` is overridden.
        Note that ``ddof=1`` will return the unbiased estimate, even if
        both `fweight` and `aweight` are specified, and ``ddof=0`` will
        return the simple average. See the notes for the details. The
        default value is ``None``.
    fweight : array_like
        1-D array of freguency weights, usually integer. The number of
        times each observation vector should be repeated.
    aweight : array_like
        1-D array of reliability weights. These relative weights are large
        for observations considered "good" and smaller for observations
        considered less "good", the details vary.

    Returns
    -------
    out : ndarray
        The covariance matrix of the variables.

    See Also
    --------
    corrcoef : Normalized covariance matrix

    Notes
    -----
    Assume that the observations are in the columns of the observation
    array `m` and let ``f = fweight`` and ``a = aweight`` for brevity. The
    steps to compute the weighted covariance are as follows::

        >>> w = f * a
        >>> v1 = np.sum(w)
        >>> v2 = np.sum(a * w)
        >>> m -= np.sum(m * w, axis=1, keepdims=True) / v1
        >>> cov = np.dot(m * w, m.T) * v1 / (v1**2 - ddof * v2)

    Note that when ``a == 1``, the normalization factor
    ``v1 / (v1**2 - ddof * v2)`` goes over to ``1 / (np.sum(f) - ddof)``
    as it should.

    Examples
    --------
    Consider two variables, :math:`x_0` and :math:`x_1`, which
    correlate perfectly, but in opposite directions:

    >>> x = np.array([[0, 2], [1, 1], [2, 0]]).T
    >>> x
    array([[0, 1, 2],
           [2, 1, 0]])

    Note how :math:`x_0` increases while :math:`x_1` decreases. The covariance
    matrix shows this clearly:

    >>> np.cov(x)
    array([[ 1., -1.],
           [-1.,  1.]])

    Note that element :math:`C_{0,1}`, which shows the correlation between
    :math:`x_0` and :math:`x_1`, is negative.

    Further, note how `x` and `y` are combined:

    >>> x = [-2.1, -1,  4.3]
    >>> y = [3,  1.1,  0.12]
    >>> X = np.vstack((x,y))
    >>> print np.cov(X)
    [[ 11.71        -4.286     ]
     [ -4.286        2.14413333]]
    >>> print np.cov(x, y)
    [[ 11.71        -4.286     ]
     [ -4.286        2.14413333]]
    >>> print np.cov(x)
    11.71

    """
```
The code block in notes can be redone as a latex version using `.. math::` if desired, but I think the numpy version is easier to read before it it typeset. Could go either way... The summary section could use further improvement as it is not very informative, or even correct (no expectation values, etc.) Might be good to find a decent reference to cite.